### ISSUE TITLE
Build errors

### ISSUE THREAD

**Issue by myint** [vmprof/vmprof-python#23]
I'm seeing the following when running `python setup.py install` on an older machine.

```
In file included from src/vmprof.c:19:
src/getpc.h:114: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘callunrollinfo’
src/getpc.h: In function ‘GetPC’:
src/getpc.h:142: error: ‘reinterpret_cast’ undeclared (first use in this function)
src/getpc.h:142: error: (Each undeclared identifier is reported only once
src/getpc.h:142: error: for each function it appears in.)
src/getpc.h:142: error: expected expression before ‘char’
src/getpc.h:143: error: ‘for’ loop initial declarations are only allowed in C99 mode
src/getpc.h:143: note: use option -std=c99 or -std=gnu99 to compile your code
src/getpc.h:143: error: ‘callunrollinfo’ undeclared (first use in this function)
$ gcc44 --version
gcc44 (GCC) 4.4.7 20120313 (Red Hat 4.4.7-1)
Copyright (C) 2010 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

I see `reinterpret_cast` in the included header file, but the file that includes it seems to be C, based on the file extension. Is this code meant to be C++ or C?

Thanks

**Comment by myint** [comment]
I guess 32-bit isn't supported. In any case, the relevant patch for this particular file is:

```diff
diff --git a/src/getpc.h b/src/getpc.h
index 999232f..b571c25 100644
--- a/src/getpc.h
+++ b/src/getpc.h
@@ -111,7 +111,7 @@ struct CallUnrollInfo {

 // -- Special case 1: linux x86, for which we have CallUnrollInfo
 #if defined(__linux) && defined(__i386) && defined(__GNUC__)
-static const CallUnrollInfo callunrollinfo[] = {
+static const struct CallUnrollInfo callunrollinfo[] = {
   // Entry to a function:  push %ebp;  mov  %esp,%ebp
   // Top-of-stack contains the caller IP.
   { 0,
@@ -139,8 +139,9 @@ inline void* GetPC(ucontext_t *signal_ucontext) {
   const int esp = signal_ucontext->uc_mcontext.gregs[REG_ESP];
   if ((eip & 0xffff0000) != 0 && (~eip & 0xffff0000) != 0 &&
       (esp & 0xffff0000) != 0) {
-    char* eip_char = reinterpret_cast<char*>(eip);
-    for (int i = 0; i < sizeof(callunrollinfo)/sizeof(*callunrollinfo); ++i) {
+    char* eip_char = (char*)(eip);
+    int i;
+    for (i = 0; i < sizeof(callunrollinfo)/sizeof(*callunrollinfo); ++i) {
       if (!memcmp(eip_char + callunrollinfo[i].pc_offset,
                   callunrollinfo[i].ins, callunrollinfo[i].ins_size)) {
         // We have a match.
```

**Comment by fijal** [comment]
Indeed, 32bit is not supported. Feel free to open a new issue with 32bit support, we want it eventually

**Comment by fijal** [comment]
32bit is tested and supported, however getpc.h is not used outside of pypy

**Comment by fijal** [comment]
we should just kill this file
